src/main/java/dev/slimevr/osc/VRCOSCHandler.java

```diff
@@ -0,0 +1,252 @@
+package dev.slimevr.osc;
+
+import com.illposed.osc.*;
+import com.illposed.osc.messageselector.OSCPatternAddressMessageSelector;
+import com.illposed.osc.transport.OSCPortIn;
+import com.illposed.osc.transport.OSCPortOut;
+import com.jme3.math.FastMath;
+import com.jme3.math.Quaternion;
+import com.jme3.math.Vector3f;
+import dev.slimevr.config.OSCConfig;
+import dev.slimevr.platform.windows.WindowsNamedPipeBridge;
+import dev.slimevr.vr.processor.HumanPoseProcessor;
+import dev.slimevr.vr.trackers.*;
+import io.eiren.util.collections.FastList;
+import io.eiren.util.logging.LogManager;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.util.List;
+
+
+/**
+ * VRChat OSCTracker documentation: https://docs.vrchat.com/docs/osc-trackers
+ */
+public class VRCOSCHandler {
+	private OSCPortIn oscReceiver;
+	private OSCPortOut oscSender;
+	private OSCMessage oscMessage;
+	private final OSCConfig config;
+	private final HMDTracker hmd;
+	private final WindowsNamedPipeBridge steamvrBridge;
+	private final HumanPoseProcessor humanPoseProcessor;
+	private final List<? extends ShareableTracker> shareableTrackers;
+	private final FastList<Float> oscArgs = new FastList<>(3);
+	private final Vector3f vec = new Vector3f();
+	private final Quaternion quatBuf = new Quaternion();
+	private final float[] floatBuf = new float[3];
+	private final boolean[] trackersEnabled;
+	private long timeAtLastOSCMessageReceived;
+	private static final long HMD_TIMEOUT = 15000;
+
+	public VRCOSCHandler(
+		HMDTracker hmd,
+		HumanPoseProcessor humanPoseProcessor,
+		WindowsNamedPipeBridge steamvrBridge,
+		OSCConfig oscConfig,
+		List<? extends ShareableTracker> shareableTrackers
+	) {
+		this.hmd = hmd;
+		this.humanPoseProcessor = humanPoseProcessor;
+		this.steamvrBridge = steamvrBridge;
+		this.config = oscConfig;
+		this.shareableTrackers = shareableTrackers;
+
+		trackersEnabled = new boolean[shareableTrackers.size()];
+
+		refreshSettings();
+	}
+
+	public void refreshSettings() {
+		// Sets which trackers are enabled and force HEAD to false
+		for (int i = 0; i < shareableTrackers.size(); i++) {
+			if (shareableTrackers.get(i).getTrackerRole() != TrackerRole.HEAD) {
+				trackersEnabled[i] = config
+					.getOSCTrackerRole(shareableTrackers.get(i).getTrackerRole(), false);
+			}
+		}
+
+		// Stops listening and closes OSC port
+		if (oscReceiver != null && oscReceiver.isListening()) {
+			oscReceiver.stopListening();
+		}
+		if (oscSender != null && oscSender.isConnected()) {
+			try {
+				oscSender.close();
+			} catch (IOException e) {
+				LogManager.warning("[VRCOSCHandler] Error closing the OSC sender: " + e);
```

Comment:
This seems like a pretty bad error, maybe log it as severe?
```suggestion
				LogManager.severe("[VRCOSCHandler] Error closing the OSC sender: " + e);
```